### PR TITLE
[CGTN] Add new extractor

### PR DESCRIPTION
**Pull request by chao813**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

### CODE REVIEW

File: yt_dlp/extractor/cgtn.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(r'<div class="news-title">(.+?)</div>', webpage, 'title')
+        download_url = self._html_search_regex(r'data-video ="(?P<url>.+m3u8)"', webpage, 'download_url')
+        thumbnail = self._html_search_regex(r'<div class="cg-player-container J_player_container" .*? data-poster="(?P<thumbnail>.+jpg)" (.*?)>', webpage, 'thumbnail', fatal=False)
+        description = self._html_search_meta(('og:description', 'twitter:description', ), webpage, fatal=False)
+        category = self._html_search_regex(r'<span class="section">\s*(.+?)\s*</span>', webpage, 'category', fatal=False)
+        datetime_str = self._html_search_regex(r'<span class="date">\s*(.+?)\s*</span>', webpage, 'datetime_str', fatal=False)
+        author = self._html_search_regex(r'<div class="news-author-name">\s*(.+?)\s*</div>', webpage, 'author', default=None, fatal=False)
```
Comment (Ashish0804) [inline yt_dlp/extractor/cgtn.py:52]: title, description and thumbnail can be obtained from `og property` using `self._og_search_thumbnail` `self._og_search_title` `self._og_search_desccription`

Comment (chao813) [reply yt_dlp/extractor/cgtn.py]: Done, didn't realize those were available in common.py

**Comment by chao813** [comment]
@Ashish0804 should this be pending review?

**Comment by Ashish0804** [comment]
Actually everything else looks fine, I'm just not sure if the timestamp extraction can be handled using an function from `utils`
I'll mark this as `pending-review` so someone else knows to take a look at aswell.

**Comment by chao813** [comment]
@Ashish0804 thanks! The timestamp extraction is working using the `unified_timestamp` function from `utils`, its passing the test cases.

File: yt_dlp/extractor/cgtn.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        download_url = self._html_search_regex(r'data-video ="(?P<url>.+m3u8)"', webpage, 'download_url')
+        thumbnail = self._og_search_thumbnail(webpage)
+        description = self._og_search_description(webpage, default=None)
+        category = self._html_search_regex(r'<span class="section">\s*(.+?)\s*</span>', webpage, 'category', fatal=False)
+        datetime_str = self._html_search_regex(r'<span class="date">\s*(.+?)\s*</span>', webpage, 'datetime_str', fatal=False)
+        author = self._html_search_regex(r'<div class="news-author-name">\s*(.+?)\s*</div>', webpage, 'author', default=None, fatal=False)
+        timestamp = None
+        if datetime_str:
+            formatted_datetime_str = datetime.datetime.strptime(datetime_str, '%H:%M, %d-%b-%Y').strftime('%Y/%m/%d %H:%M')
+            timestamp = unified_timestamp(formatted_datetime_str) - 8 * 3600
```
Comment (pukkandan) [inline yt_dlp/extractor/cgtn.py:56]: if `unified_timestamp` doesn't directly work, add the format to `utils.DATE_FORMATS`

File: yt_dlp/extractor/cgtn.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                'title': 'China, Indonesia vow to further deepen maritime cooperation',
+                'thumbnail': r're:^https?://.*\.png$',
+                'description': 'China and Indonesia vowed to upgrade their cooperation into the maritime sector and also for political security, economy, and cultural and people-to-people exchanges.',
+                'author': 'CGTN',
+                'category': 'China',
+                'timestamp': 1622950200,
+                'upload_date': '20210606',
+            }
+        }
+    ]
```
Comment (pukkandan) [inline yt_dlp/extractor/cgtn.py:40]: add `skip_download` to the tests

File: yt_dlp/extractor/cgtn.py (hunk 2 above)
Comment (chao813) [reply yt_dlp/extractor/cgtn.py:56]: @pukkandan The date format is '%H:%M, %d-%b-%Y' and adding it to the `utils.DATE_FORMATS` doesn't work. `unified_timestamp` doesn't parse it correctly.

Comment (pukkandan) [reply yt_dlp/extractor/cgtn.py:62]: found the problem https://github.com/ytdl-org/youtube-dl/issues/29948

Comment (chao813) [reply yt_dlp/extractor/cgtn.py:62]: I know, I think dirkf's PR should've fixed it right? What should the next steps be for me.

**Pull request merged**